### ISSUE TITLE
Axe puppeteer hangs when JS is disabled

### ISSUE THREAD

**Issue by michaelchiche** [dequelabs/axe-core-npm#529]
Product: puppeteer

Expectation: I tried to make it work with Javascript disabled on the page, and the analyze function just hangs, i dont know if this is wanted/expected behavior of if there is a way to make it work, or improve the error handling?

Actual: It hangs with Js disabled on the page
Motivation: from what i know, we only read the dom and see if the elements answer some rules. why do we need js to make it work. It also helps save some time because it renders the page from the server only.

```ts
import { AxePuppeteer } from '@axe-core/puppeteer';
import puppeteer from 'puppeteer';

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.setBypassCSP(true);
  await page.setJavaScriptEnabled(false);

  await page.goto('https://dequeuniversity.com/demo/mars/');
  console.log('before results');

  const results = await new AxePuppeteer(page).analyze();
  console.log(results);

  await page.close();
  await browser.close();
})();

```

<pre><code>
@axe-core/puppeteer@4.4.3

- Node version: 16.15.1
- Platform:  Linux
</code></pre>

**Comment by michaelchiche** [comment]
I read the code a little bit, it looks like the package runs JS inside the page, so maybe it should be documented?

**Comment by straker** [comment]
Yes, all of our integrations run axe-core on the page you are analyzing, which requires JavaScript to be enabled on the page. We'll look at improving our documentation to better describe things.

**Comment by padmavemulapati** [comment]
As we are running javascript enabled scripts, issue trying is expected.

**Comment by padmavemulapati** [comment]
Validated with the latest `axe-core/puppeteer@ ` , with the caveat, `await page.setJavaScriptEnabled(false);` we are unable to do execution.



![Image](https://user-images.githubusercontent.com/47409579/208893587-fc039871-6b17-4178-a846-ed218f3b7fba.png)

Enhancement suggested:
It is good if it display any warning message and quit from the execution
